internal/provider/application_resource_test.go

```diff
@@ -69,15 +70,47 @@ func TestAccApplicationResource(t *testing.T) {
 	})
 }
 
-func testAccApplicationResource(name string, update string) string {
+func TestAccApplicationResourceMoveOrganization(t *testing.T) {
+	appName := `TFACC` + acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	organizationIdRegex, _ := regexp.Compile(`^[a-z0-9]{32}$`)
+	resourceName := "sonatypeiq_application.test"
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
```

Comment:
Could add a test to confirm appropriate error when attempting to Move and Application to either a non-existent Organisation or lack or permission @grahamhar ?

Reply:
Hi @madpah,

I see that behaviour being more the concern of the underlying SDK, then we simply handle any error types [here](https://github.com/sonatype-nexus-community/terraform-provider-sonatypeiq/pull/27/files#diff-d488c131a16336ed06a2ce3793408b36f68ad766c5ad1f1e6f606af43d550863R211) and propagate to the user. 

Also I don't believe we test for permissions errors or bad parent org in app or organisation creation so if we did go down this road I guess we would want to add the same test approaches in those areas too?

Reply:
I would be interested in how you would see that test being implemented I've not come across that type of test on other providers so it would be great to get a reference to an example

Reply:
See https://github.com/sonatype-nexus-community/terraform-provider-sonatyperepo/blob/35a32b41b597523d293c6a056f99e34d77737165/internal/provider/repository/repository_maven_group_resource_test.go#L42 for an example @grahamhar

Reply:
Added test step for unknown organisation, I think trying to cover a scenario
for not authorised would be hard as we would need to make sure we had an organisation available that the test runner didn't have access to. I think having the one test shows that any errors from the SDK should be bubbled up through the provider/resource.